modyn/common/trigger_sample/trigger_sample_storage.py

```diff
@@ -294,3 +294,35 @@ def _get_num_samples_in_file(self, file_path: Path) -> int:
             file_path (str): File path to parse.
         """
         return np.load(file_path, allow_pickle=False, fix_imports=False, mmap_mode="r").shape[0]
+
+    def get_file_path(self, pipeline_id: int, trigger_id: int, partition_id: int, worker_id: int) -> Path:
+        return Path(self.trigger_sample_directory) / f"{pipeline_id}_{trigger_id}_{partition_id}_{worker_id}.npy"
+
+    def get_trigger_num_data_partitions(self, pipeline_id: int, trigger_id: int) -> int:
+        # each file follows the structure {pipeline_id}_{trigger_id}_{partition_id}_{worker_id}
+
+        # here we filter the files belonging to this pipeline and trigger
+        this_trigger_files = list(
+            filter(
+                lambda file: file.startswith(f"{pipeline_id}_{trigger_id}_"),
+                os.listdir(self.trigger_sample_directory),
+            )
+        )
```

Comment:
can be put in function `_get_files_for_trigger` and then re-used in `clean_trigger_data`

Reply:
Sure, done